tests/unit/s2n_resume_test.c

```diff
@@ -760,19 +810,23 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_set_server_early_data_context(conn, test_early_data_context, sizeof(test_early_data_context)));
             EXPECT_MEMCPY_SUCCESS(conn->application_protocol, test_app_proto, sizeof(test_app_proto));
             conn->actual_protocol_version = S2N_TLS13;
+            conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
 
             DEFER_CLEANUP(struct s2n_stuffer output = { 0 }, s2n_stuffer_free);
             EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&output, 0));
             struct s2n_ticket_fields ticket_fields = { .ticket_age_add = 1, .session_secret = test_master_secret };
 
             EXPECT_SUCCESS(s2n_encrypt_session_ticket(conn, &ticket_fields, &output));
+            conn->client_ticket_to_decrypt = output;
+            EXPECT_SUCCESS(s2n_decrypt_session_ticket(conn));
+
+            EXPECT_EQUAL(s2n_stuffer_data_available(&conn->client_ticket_to_decrypt), 0);
 
-            uint32_t expected_size = FIXED_ENCRYPTED_TICKET_SIZE + test_master_secret.size
-                        + sizeof(uint16_t) /* size of context size */
-                        + sizeof(test_early_data_context)
-                        + sizeof(uint8_t) /* application protocol size */
-                        + strlen(test_app_proto);
-            EXPECT_EQUAL(expected_size, s2n_stuffer_data_available(&output));
+            /* Check decryption was successful */
+            struct s2n_psk *psk = NULL;
+            EXPECT_OK(s2n_array_get(&conn->psk_params.psk_list, 0, (void**) &psk));
+            EXPECT_NOT_NULL(psk);
+            EXPECT_EQUAL(psk->hmac_alg, s2n_tls13_aes_128_gcm_sha256.prf_alg);
```

Comment:
Since you're specifically testing "with early data" here, can you verify an early data field?